### PR TITLE
Bug 1872981: Add basic functionality tests for selecting multiple storage classes

### PR DESCRIPTION
**Pull request by afreen23**
Screenshot from 2020-08-20 16-53-47

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    devicesCount: null,
+  };
+  beforeAll(init);
+  afterAll(teardown);
+  describe('Add capacity with a new storage class having EBS as provisioner', async () => {
+    const { name: scName } = mockSC.metadata;
+    let index: number;
+    let deviceSets: DeviceSet[];
+    beforeAll(async () => {
+      const yaml: K8sResourceKind = fetchStorageClusterYAML();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts:64]:
```suggestion
      const yaml: K8sResourceKind = fetchStorageClusterJSON();
```

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      beforeCapacityAddition.deviceSets = yaml.spec.storageDeviceSets.length;
+      await addCapacity(yaml.metadata.uid, scName);
+      const latestYaml: K8sResourceKind = fetchStorageClusterYAML();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts:67]:
```suggestion
      const latestYaml: K8sResourceKind = fetchStorageClusterJSON();
```

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    it('Device count is 1 in the new device set', () => expect(deviceSets[index].count).toBe(1));
+    it('Osd portability is enabled in the new device set', () =>
+      expect(deviceSets[index].portable).toBe(true));
+  });
+  describe('Add capacity with an existing storage class having EBS as provisioner', async () => {
+    const { name: scName } = mockSC.metadata;
+    let latestDeviceSets: DeviceSet[];
+    let latestIndex: number;
+    beforeAll(async () => {
+      const yaml: K8sResourceKind = fetchStorageClusterYAML();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts:82]:
```suggestion
      const json: K8sResourceKind = fetchStorageClusterJSON();
```

File: frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts [hunk 4]
```diff
@@ -1,10 +1,16 @@
-export const storageClass = {
+export const mockNoProvisionerSC = {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts:1]:
```suggestion
export const NoProvisionerSC = {
```
No need to add mock in front as it's all inside the mock folder.

File: frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts [hunk 5]
```diff
@@ -1,10 +1,16 @@
...
-  metadata: {
-    name: 'test-sc',
-    namespace: 'openshift-storage',
-  },
+  metadata: { name: 'mock-no-prov-sc' },
   provisioner: 'kubernetes.io/no-provisioner',
   reclaimPolicy: 'Delete',
 };
+
+export const mockSC = {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts:9]: Same as above.

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+import { OCS_INTERNAL_CR_NAME } from '../../../src/constants';
+import { getCurrentDeviceSetIndex } from '../../../src/utils/add-capacity';
+import { mockSC, mockNoProvisionerSC } from '../../mocks/storage-class';
+import {
+  goToInstalledOperators,
+  currentACSelector,
+  clickKebabAction,
+} from '../../views/add-capacity.view';
+import { NS } from '../../utils/consts';
+import { appHost } from '@console/internal-integration-tests/protractor.conf';
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts:15]: Nit: move it above relative imports.

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+const fetchStorageClusterYAML = () => {
+  const options: ExecFileOptionsWithStringEncoding = {
+    encoding: 'utf8',
+  };
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts:20]: Not required default is utf8

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts (hunk 7 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts:20]: Default is 'Buffer'. https://nodejs.org/api/child_process.html#child_process_child_process_execsync_command_optio

File: frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts (hunk 4 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts:1]: When I see `mockNoProvisionerSC` somewhere I can make out that it is a mock storage class.

Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts:1]: Let's follow the pattern that we already have set.

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts (hunk 7 above)
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts:20]: Ack

File: frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts [hunk 8]
```diff
@@ -1,10 +1,16 @@
-export const storageClass = {
+export const mockNoProvisionerSC = {
   apiVersion: 'storage.k8s.io/v1',
   kind: 'StorageClass',
-  metadata: {
-    name: 'test-sc',
-    namespace: 'openshift-storage',
-  },
+  metadata: { name: 'mock-no-prov-sc' },
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts:4]:
```suggestion
  metadata: { name: 'no-prov-sc' },
```

File: frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts [hunk 9]
```diff
@@ -1,10 +1,16 @@
...
   provisioner: 'kubernetes.io/no-provisioner',
   reclaimPolicy: 'Delete',
 };
+
+export const mockSC = {
+  apiVersion: 'storage.k8s.io/v1',
+  kind: 'StorageClass',
+  metadata: { name: 'mock-ebs-sc' },
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts:12]:
```suggestion
  metadata: { name: 'ebs-sc' },
```

**Review by bipuladh — COMMENTED** [review]
Please rename the title. `Scenarios` is not clear.

**Comment by afreen23** [comment]
/hold 
looking into failing tests

File: frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts (hunk 8 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts]: Lets have `test-no-prov-sc`

File: frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts (hunk 9 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts]: Lets have `test-ebs-sc` to differentiate it as a mock stoarge class for testing pupose

File: frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts (hunk 4 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/integration-tests/mocks/storage-class.ts]: Renamed the test classes to be prefixed with "test" .

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts (hunk 3 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts]: ack

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts (hunk 2 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/integration-tests/tests/3-tests/multiple-storage-class-selection.scenario.ts]: ack

**Pull request merged**